internal/statsd.go

```diff
@@ -15,3 +21,90 @@ type StatsdClient interface {
 	Flush() error
 	Close() error
 }
+
+type Stat struct {
+	Name  string
+	Kind  string
+	Value interface{} //Really, it can only be a "number" type, but wasn't sure if we want to enforce that here yet
+	Tags  []string
+	Rate  float64
+}
+
+// StatsCarrier collects stats on its contribStats channel and submits them to the Datadog agent via a statsd client
+type StatsCarrier struct {
+	contribStats chan Stat
+	statsd       StatsdClient
+	stop         chan struct{}
+	wg           sync.WaitGroup
+	stopped uint64
+}
+
+func NewStatsCarrier(statsd StatsdClient) *StatsCarrier {
+	return &StatsCarrier{
+		contribStats: make(chan Stat),
+		statsd:       statsd,
+		stopped: 1,
+	}
+}
+func (sc *StatsCarrier) Start() {
+	if atomic.SwapUint64(&sc.stopped, 0) == 0 {
+		// already running
+		log.Warn("(*StatsCarrier).Start called more than once. This is likely a programming error.")
```

Comment:
I'd consider deleting this, since it doesn't provide much value. LMK if you feel it's needed for something specific though, or if it's just for your own debugging right now.

Reply:
I was following the design on the [Processor](https://github.com/DataDog/dd-trace-go/blob/main/internal/datastreams/processor.go#L181) for ensuring that we never attempt to close the `stop` channel once it's already been closed - where 0 represents that the `stop` channel has not yet been closed, and it's changed to 1 (and also checked for a 1 value) in the StatsCarrier.Stop() method